test/functional/tag_controller_test.rb

```diff
@@ -48,6 +48,25 @@ def setup
     assert_response :success
   end
 
+  test 'normal users cannot add the locked tag' do
+    UserSession.create(users(:bob))
+    post :create,
+         params: {
+            name: 'locked',
+            nid: nodes(:one).nid
+         }
+    assert_equal 'Error: only admins can lock pages.', assigns[:output][:errors][0]
+  end
+
+  test 'admin and moderators can add the locked tag' do
+    UserSession.create(users(:admin))
+    UserSession.create(users(:moderator))
```

Comment:
Thanks for adding this test @Manasa2850!

Is the UserSession being used from line 64-68? It seems to me that line 64-68 is interacting with the models directly and looks more like a unit test. There is no post request using line 62 or 63...

This test suite can run without the UserSession, but I wonder if that makes sense since this is a controller test? Which is testing the actions?

My suggestion would be to split this test into 2, one for the admin and another for the moderator, where you can use the UserSession and hit the endpoint after, similar to what you have done in line 51

Reply:
Sure @RuthNjeri I'll split it into 2 tests. I had combined them into one test to avoid repetition but what you mentioned about using UserSession makes more sense. Thank you!